### PR TITLE
Some typos and remarks

### PR DESCRIPTION
**Pull request by ARPL-c-63h**
Used {::comment}{:/comment} for elaborations.

### CODE REVIEW

File: haskell-for-readers.md [hunk 1]
```diff
@@ -759,6 +768,9 @@ From this example, we learn that
  * the function arrow can just as well occur inside an argument, namely when an argument itself is a function.
 
 But what is this type `t`? There is not, actually, a type called `t`. Instead, this is a *type variable*, meaning that the function `twice` can be used with any type. Any lower-case identifier in a type is a type variable (not just `t`), and concrete types are always upper-case.
+{::comment}
+That is not entirely accurate. It's whether they /start/ with a lower- or upper-case letter.
```
Comment (nomeata) [inline haskell-for-readers.md]: Right, that’s of course what I try to say. Feel free to propose a better wording :-)

File: haskell-for-readers.md [hunk 2]
```diff
@@ -1102,7 +1114,7 @@ There are four: `Nothing`, `Just Nothing`, `Just (Just False)` and `Just (Just T
 
 ### `Either` ☆
 
-With maybe we can express “one or none”. Sometimes we want “one or another” type. For this, the standard library provides
+With Maybe we can express “one or none”. Sometimes we want “one or another” type. For this, the standard library provides
```
Comment (nomeata) [inline haskell-for-readers.md]:
```suggestion
With `Maybe` we can express “one or none”. Sometimes we want “one or another” type. For this, the standard library provides
```

File: haskell-for-readers.md [hunk 3]
```diff
@@ -1385,7 +1400,9 @@ rle stop =
   in NeedInput start
 ```
 In `start` we can now access `stop` just fine. If `start` and `count` were not local functions, then we would have to add `stop` as an explicit parameter to *both* local functions, significantly cluttering the code with administrative details.
-
+{::comment}
+Personally I found the rle example not easy to follow. What you tried to explain in each section was fine. But the usage of rle was not immediately clear. I guess this is an example of where an "interactive lecture" is the preferred presentation form.
```
Comment (nomeata) [inline haskell-for-readers.md]: Yeah. At least I hope it was better in my lecture :-)

File: haskell-for-readers.md [hunk 4]
```diff
@@ -2670,8 +2687,27 @@ What can we do with that freedom? Create more instances!
...
+At a high level, the main advantage of an applicative computation over a monadic one is that the *structure of the computation* is static, and can be known ahead of time. This is not the case in a monadic computation, because the second argument to `(>>=)` is an opaque function, and only once a concrete `a` was extracted from the first argument do we know what kind of computation is produced by the second.
 
 As a concrete example, consider a library that provides a type constructor `AParser` that is an instance of `Applicative`, but *not* of `Monad`. You write a parser `p :: AParser a` using this interface. Then the library can “run” this parser without actual input, and learn everything it *would* do. It could use that to provide a function `describe :: AParser a -> String` that produces a grammar for the given parser, which you can use as documentation, and thus these can never go out of sync.
 
 
+{::comment}
+Solution 16
+  p59: ... and dedicated data types with more *speaking* names ...
+       [In German "sprechend", like in Dutch "sprekend", would be fine, but
+        I think for English something like "descriptive" would be better.]
```
Comment (nomeata) [inline haskell-for-readers.md]: Descriptive sounds good, bought :-)

File: haskell-for-readers.md [hunk 5]
```diff
@@ -2670,8 +2687,27 @@ What can we do with that freedom? Create more instances!
...
+
+[On pages 61 and 62 the order of the solutions is: 24, 27, 25, 26, 28.]
+
+
+Solution 26
+  p62: "m" sounds like it starts with a vowel, so both "a m a" should be
+       *an m a*.
+Solution 28
+  p62: ... exactly the operations needed to [*-to*] implement pure ...
+{:/comment}
```
Comment (nomeata) [inline haskell-for-readers.md]: The solutions are actually also part of this `.md` files, so you can change them directly.

**Review by nomeata — APPROVED** [review]
Amazing work, thanks a lot! Happy with all your changes.

If you remove the comments I’ll merge it.

**Comment by nomeata** [comment]
Thanks a lot, this is great!

**Comment by ARPL-c-63h** [comment]
Joachim,
I have directed quite a number of people to Haskell for Readers. Yesterday I spoke one of them, an auditor, and he said that it had been very helpful. Always nice to know. : )
-- CAJ

**Comment by nomeata** [comment]
Nice to hear! :-)

**Pull request merged**